tests/network/vmi_istio.go

```diff
@@ -0,0 +1,244 @@
+		tests.BeforeAll(func() {
+			if !istioServiceMeshDeployed() {
+				Skip("Istio service mesh is required for service-mesh tests to run")
+			}
+
+			tests.BeforeTestCleanup()
+			virtClient, err = kubecli.GetKubevirtClient()
+			tests.PanicOnError(err)
+
+			vmiWithExplicitPorts = newRandomCirrosVMIWithIstioProxyAndEphemeralDiskAndUserDataAndAndServiceAccountDiskAndNetworkAndInterface(
+				"default",
+				vmiExplicitPortsAppSelector,
+				"true",
+				"#!/bin/bash\necho 'hello'\n",
+				&defaultNetwork,
+				&interfaceWithExplicitPorts,
+			)
+			vmiWithDefaultPorts = newRandomCirrosVMIWithIstioProxyAndEphemeralDiskAndUserDataAndAndServiceAccountDiskAndNetworkAndInterface(
+				"default",
+				vmiDefaultPortsAppSelector,
+				"true",
+				"#!/bin/bash\necho 'hello'\n",
+				&defaultNetwork,
+				&defaultInterface,
+			)
+			vmiWithExplicitPortsService := newServiceWithAppSelector(fmt.Sprintf("%s-service", vmiWithExplicitPorts.Name), vmiWithExplicitPorts.Namespace, vmiExplicitPortsAppSelector)
+			vmiWithDefaultPortsService := newServiceWithAppSelector(fmt.Sprintf("%s-service", vmiWithDefaultPorts.Name), vmiWithDefaultPorts.Namespace, vmiDefaultPortsAppSelector)
+
+			addLabelToNamespace(vmiWithExplicitPorts.Namespace, istioInjectNamespaceLabel, "enabled")
+			for _, svc := range []corev1.Service{*vmiWithExplicitPortsService, *vmiWithDefaultPortsService} {
+				_, err = virtClient.CoreV1().Services(svc.Namespace).Create(context.Background(), &svc, metav1.CreateOptions{})
+				Expect(err).ShouldNot(HaveOccurred())
+			}
+			for _, networkVMI := range []*v1.VirtualMachineInstance{vmiWithExplicitPorts, vmiWithDefaultPorts} {
+				_, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(networkVMI)
+				Expect(err).ToNot(HaveOccurred())
+			}
+
+			// Wait for VMIs to become ready
+			vmiWithExplicitPorts = tests.WaitUntilVMIReady(vmiWithExplicitPorts, libnet.WithIPv6(console.LoginToCirros))
+			vmiWithDefaultPorts = tests.WaitUntilVMIReady(vmiWithDefaultPorts, libnet.WithIPv6(console.LoginToCirros))
+
+			for _, vmi := range []*v1.VirtualMachineInstance{vmiWithExplicitPorts, vmiWithDefaultPorts} {
+				tests.StartHTTPServer(vmi, svcDeclaredTestPort)
+				tests.StartHTTPServer(vmi, svcUndeclaredTestPort)
+			}
+		})
+
+		table.DescribeTable("should be able to reach", func(destination string, port int) {
+			var srcVMI, dstVMI *v1.VirtualMachineInstance
+			switch destination {
+			case "vmiWithExplicitPorts":
+				srcVMI = vmiWithDefaultPorts
+				dstVMI = vmiWithExplicitPorts
+			case "vmiWithDefaultPorts":
+				srcVMI = vmiWithExplicitPorts
+				dstVMI = vmiWithDefaultPorts
+			}
+			dstVMIAddress := dstVMI.Status.Interfaces[0].IP
+
+			By(fmt.Sprintf("checking that service running on port %d can be reached", port))
+			cmdCheck := fmt.Sprintf("curl %s:%d", dstVMIAddress, port)
+			err = console.SafeExpectBatch(srcVMI, []expect.Batcher{
```

Comment:
`console.RunCommand` can be used here.

Reply:
console.RunCommand is so much faster, thanks! Done